From 09949eadbb122b4c8dd1813b145c0ab9a8de69fe Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Fri, 31 Mar 2017 14:04:27 +0800
Subject: [PATCH 54/62] support initramfs boot and lvm auto partition

- Create fake initramfs to cheat grub-mkconfig which
  could generate grub.cfg that have initrd setting.

- Set lvm as auto partition.

- Make sure temp dir existed which required by dracut while live install.

- Improve condition of updating kernel version list.

Upstream-Status: Inappropriate [wrlinux specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 pyanaconda/constants.py          | 3 +--
 pyanaconda/install.py            | 6 ++++++
 pyanaconda/packaging/__init__.py | 2 +-
 3 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/pyanaconda/constants.py b/pyanaconda/constants.py
index 8986a64..af240d3 100644
--- a/pyanaconda/constants.py
+++ b/pyanaconda/constants.py
@@ -183,8 +183,7 @@ SCREENSHOTS_TARGET_DIRECTORY = "/root/anaconda-screenshots"
 # cmdline arguments that append instead of overwrite
 CMDLINE_APPEND = ["modprobe.blacklist", "ifname"]
 
-#DEFAULT_AUTOPART_TYPE = AUTOPART_TYPE_LVM
-DEFAULT_AUTOPART_TYPE =  AUTOPART_TYPE_PLAIN
+DEFAULT_AUTOPART_TYPE = AUTOPART_TYPE_LVM
 
 import logging
 LOGLVL_LOCK = logging.DEBUG-1
diff --git a/pyanaconda/install.py b/pyanaconda/install.py
index 7b72fe0..2fca486 100644
--- a/pyanaconda/install.py
+++ b/pyanaconda/install.py
@@ -206,6 +206,9 @@ def doInstall(storage, payload, ksdata, instClass):
                                                             wait_for_entropy=entropy_wait_clbk)
 
     turnOnFilesystems(storage, mountOnly=flags.flags.dirInstall, callbacks=callbacks_reg)
+    # Make sure temp dir existed, dracut requires it while live install
+    for subdir in ["log", "tmp"]:
+        iutil.mkdirChain(iutil.getSysroot() + "/var/volatile/%s" % subdir)
 
     # Run %pre-install scripts with the filesystem mounted and no packages
     with progress_report(_("Running pre-installation scripts")):
@@ -275,6 +278,9 @@ def doInstall(storage, payload, ksdata, instClass):
 
     # Do bootloader.
     if willInstallBootloader:
+        for kernel in payload.kernelVersionList:
+            open(iutil.getSysroot()+"/boot/initramfs-%s.img" % kernel, "w").write("\n")
+
         with progress_report(_("Installing boot loader")):
             writeBootLoader(storage, payload, instClass, ksdata)
 
diff --git a/pyanaconda/packaging/__init__.py b/pyanaconda/packaging/__init__.py
index 683b2a0..72a854c 100644
--- a/pyanaconda/packaging/__init__.py
+++ b/pyanaconda/packaging/__init__.py
@@ -438,7 +438,7 @@ class Payload(object):
 
     @property
     def kernelVersionList(self):
-        if not self._kernelVersionList:
+        if not self._kernelVersionList or not self._kernelVersionList[0]:
             self._updateKernelVersionList()
 
         return self._kernelVersionList[0]
-- 
2.7.4

